### ISSUE TITLE
Implement configurable allowed soils

### ISSUE THREAD

**Issue by juraj-hrivnak** [juraj-hrivnak/Greenery#22]
Syntax:
```
{
  "soil": [
    "material:<material>",
    "block:<resource_location>",
    "block:<resource_location> | <blockstates>..."
  ]
}
```
Example:
```
{
  "soil": [
    "material:grass",
    "block:minecraft:sand",
    "block:minecraft:stone | variant=granite"
  ]
}
```

**Comment by juraj-hrivnak** [comment]
Halfway implemented in v5.0

**Comment by juraj-hrivnak** [comment]
Fixed in a93d434
